meta.go

```diff
@@ -102,6 +102,59 @@ func (o *Object) Type() string {
 	return o.typ
 }
 
+// GetString looks up the property with the given key, type asserts it as a
+// string and returns it.
+func (o *Object) GetString(key string) (string, error) {
+	v, err := o.Get(key)
+	if err != nil {
+		return "", err
+	}
+	s, ok := v.(string)
+	if !ok {
+		return "", fmt.Errorf("key %q has type %T, not string", key, v)
+	}
+	return s, nil
+}
+
+// GetLink looks up the property with the given key, type asserts it as a
+// link and returns it.
+func (o *Object) GetLink(key string) (*format.Link, error) {
+	v, err := o.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	l, ok := v.(*format.Link)
+	if !ok {
+		return nil, fmt.Errorf("key %q has type %T, not *format.Link", key, v)
+	}
+	return l, nil
+}
+
+// GetList looks up the property with the given key, type asserts it as a
+// generic list and returns it.
+func (o *Object) GetList(key string) ([]interface{}, error) {
+	v, err := o.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	l, ok := v.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("key %q has type %T, not []interface{}", key, v)
+	}
+	return l, nil
+}
+
+// Get returns the property with the given key.
+func (o *Object) Get(key string) (interface{}, error) {
+	v, rest, err := o.node.Resolve([]string{key})
```

Comment:
`rest` represents the "rest" of the path which is left after finding a link whilst traversing the path.

If you have an object graph like this:

```
a = { "a": { "/": "<hash of b>" }
b = { "b": { "/": "<hash of c>" }
c = { "c": { "/": "<hash of d>" }
```

Then doing `a.Resolve([]string{"a", "b", "c"})` will return `<hash of b>` with rest being `[]string{"b", "c"}` meaning it found `a` but you need to resolve the rest starting from `b` once you have retrieved it.

We will improve this API in the future (I've already discussed future modifications of the IPLD CBOR lib), but for now this is what we have.